enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,330 @@
+[assisted-agent](https://github.com/openshift/assisted-installer-agent#readme))
+of the Assisted and Multi-Cluster Engine (MCE) installation methods. The
+installer will be usable in both a fully-automated workflow, where
+configuration is provided upfront, and an interactive workflow where the user
+can interact with a GUI similar to the hosted Assisted installation service.
+
+## Motivation
+
+Every on-premises user has a first cluster ("Cluster 0"), and many of them will
+be installed in a disconnected environment. While powerful tools exist for
+creating multiple on-premises clusters (Hive, HyperShift, MCE), they all run in
+a cluster themselves, which must somehow be provisioned without these tools.
+
+Each of the current installation options has limitations that make it
+unpalatable to a subset of users. IPI is highly opinionated, and suffers from
+providing limited visibility of configuration problems in on-premises
+environments. UPI exposes users to the full complexity of OpenShift to the user
+with little guidance. The (hosted) Assisted service is not available for
+IPv6-only (until Quay.io supports IPv6) or disconnected environments. For some
+users, none of these options is optimal.
+
+On-premises users need a simple way to deploy their first cluster, even in a
+disconnected environment, using their own automation to boot hosts.
+
+### Goals
+
+- Install clusters in fully-disconnected environments.
+- Make use of assisted installation components to pre-validate the installation
+  environment.
+- Perform reproducible cluster builds from configuration artifacts, as well as
+  interactive installation.
+- Require no machines in the cluster environment other than those that are to
+  become part of the cluster.
+- Install clusters with single-node, compact, and high-availability topologies.
+- Be agnostic to the tools used to provision (virtual) machines so that users
+  can use their own (possibly pre-existing) tooling for provisioning.
+- Support install configurations with platform `none`, `baremetal`, or
+  `vsphere`.
+- Run on either Linux (any distro) or MacOS, in either x86-64 or arm64
+  architectures.
+- Install to machines with either x86-64 or arm64 architectures, independent of
+  the architecture the CLI tool runs on.
+
+### Non-Goals
+
+- Replace the IPI installer in any capacity.
+- Automate booting of the image on (virtual) machines.
+- Support install configurations for cloud-based platforms.
+- Provide an interactive API to drive automated deployments.
+- Operate in disconnected environments without the need to separately mirror
+  the release payload to a registry that is usable from the installed cluster.
+- Provide a mechanism to add and remove nodes from the cluster after it is
+  installed.
+- Generate image formats other than ISO.
+- Create minimal ISO images with a separate rootfs.
+- Run on Windows.
+
+## Proposal
+
+A command-line tool will enable users to build a custom RHCOS image, initially
```

Comment:
I think this paragraph would be easier to follow if it were more linear.
* An openshift release needs to be available in a local mirror
* The tool gets the base RHCOS image from that release
* The tool creates a new ISO
* The new ISO gets booted (fixes some current ambiguity on line 95 about which is "this" ISO that gets booted)
* Other required container images get pulled from the registry after boot